ui/apps/platform/src/services/CollectionsService.ts

```diff
@@ -0,0 +1,234 @@
+import qs from 'qs';
+
+import { ListDeployment } from 'types/deployment.proto';
+import { SearchFilter, ApiSortOption } from 'types/search';
+import { getListQueryParams, getRequestQueryStringForSearchFilter } from 'utils/searchUtils';
+import { CancellableRequest, makeCancellableAxiosRequest } from './cancellationUtils';
+import axios from './instance';
+import { Empty, Pagination } from './types';
+
+export const collectionsBaseUrl = '/v1/collections';
+export const collectionsCountUrl = '/v1/collections/count';
+export const collectionsDryRunUrl = '/v1/collections/dryrun';
+export const collectionsAutocompleteUrl = '/v1/collections/autocomplete';
+
+type CollectionResource = 'Cluster' | 'Namespace' | 'Deployment';
```

Comment:
Remembering the discussion which motivated change from resource sets to collections:

Is it worthwhile to ask whether there is a meaningful alternative to **Resource** in collection API?

https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/

> A resource is an endpoint in the Kubernetes AP that stores a collection of API objects of a certain kind; for example, the built-in pods resource contains a collection of Pod objects.

> A custom resource is an extension of the Kubernetes API that is not necessarily available in a default Kubernetes installation. It represents a customization of a particular Kubernetes installation. However, many core Kubernetes functions are now built using custom resources, making Kubernetes more modular.

Reply:
Here is a thought:
1. Replace `CollectionResource` with `SelectorEntityType`
2. Replace `resourceSelectors: ResourceSelector[]` with `collectionSelectors: CollectionSelector[]` which implies discussion with core workflows team

Reply:
Sure I can bring this up for discussion during our weekly sync and refactor after if needed.